specifyweb/frontend/js_src/lib/components/Statistics/StatItems.tsx

```diff
@@ -133,37 +141,32 @@ function BackEndItem({
   readonly onClick: (() => void) | undefined;
   readonly onRemove: (() => void) | undefined;
   readonly onItemRename: ((newLabel: string) => void) | undefined;
-  readonly onItemValueLoad:
-    | ((value: number | string, itemLabel: string) => void)
-    | undefined;
+  readonly onItemValueLoad: ((value: number | string) => void) | undefined;
 }): JSX.Element {
-  const [count] = useAsyncState<number | string | undefined>(
-    React.useCallback(
+  const shouldFetch =
```

Comment:
This is a good way to do it as it prevent needless calls to useEffect.

Calls to useEffect may cause react to re-render even if setState wasn't called (don't remember the exact reason why, but I think they needed to do it to ensure consistency), thus it might be worth handling conditions outside useEffect, if it means useEffect has fewer dependencies.

But also, I think it's just cleaner (personal opinion)